### DISCUSSION TITLE
Display test name before running it

### DISCUSSION THREAD

**Discussion by jhagege** [pytest-dev/pytest-forked#51] [Q&A] [answered]
Hi, I am sometimes having a hard time troubleshooting failing tests stuck in an infinite loop. 
The reason is that output is being printed only if test finished (fail or success), but if it is stuck, then it is very hard to identify which test is stuck.
Could you please suggest a pattern to print the test name to STDOUT / STDERR (on a CI server for example) before running any test ? 
Thanks much.

**Comment by jhagege** [comment]
Also, are you aware of any pattern to retrieve logs from an ongoing test (stuck test for example) ?

**Reply by webknjaz** [reply]
What I've been doing lately with GitHub Actions workflows CI/CD is that I've looked at how much time the test step normally takes and applied `timeout-minutes` so that GitHub would kill the step. Then pytest would show some info in the failure details. Plus it wouldn't get stuck for like 6 hours waiting for the runner to finally kill the job...

**Comment by webknjaz** [comment, marked as answer]
I believe increasing the pytest verbosity with `-vvvvv` should solve your problem.

**Reply by jhagege** [reply]
Thanks much, will check
